### PR TITLE
[FIX] account: search's `filter_domain` in the currencies list view 

### PR DESCRIPTION
**Pull request by rigr-odoo**
To reproduce the bug:
- Install the Invoicing/Accounting app
- Go to the Settings
- Go to the Currencies
- Search for a currency (e.g. Euro)
- An exception is raised (InvalidDomainError: Invalid domain AST)

This error is thrown because the expected type of the `filter_domain` is a list, and not a tuple.

This PR fixes this problem by converting the `filter_domain` into a list.

task-id 2948653

### CODE REVIEW

**Review by xmo-odoo — CHANGES_REQUESTED** [review]
The PR and commit messages should specify what the error is, and explain what the diagnosed bug is. The OPW link is not at all helpful, and is not even an OPW (an opw is a client ticket).

This also seems like a client-side regression or at least change, there are tuple `filter_domain` dating back to 2016, and this one was merged in May.

This should be taken up to the JS framework team, if they simply forgot that case the fix is to add it back, if they have decided to stop supporting such filter domains, either a migration or a handling of the error needs to be implemented, to not make views with such filter domains unusable on update.

In the latter case, the tuple filter_domains in odoo/addons/base/views/ir_translation_views.xml also need to be fixed.

**Comment by rigr-odoo** [comment]
> The PR and commit messages should specify what the error is, and explain what the diagnosed bug is. The OPW link is not at all helpful, and is not even an OPW (an opw is a client ticket).

Indeed, it is fixed.

> This also seems like a client-side regression or at least change, there are tuple `filter_domain` dating back to 2016, and this one was merged in May.
This should be taken up to the JS framework team, if they simply forget that case the fix is to add it back, if they have decided to stop supporting such filter domains, either a migration or a handling of the error needs to be implemented, to not make views with such filter domains unusable on update.

I only see three occurrences of this problem (this one and two in `ir_translation_views.xml`). I will check what should be done with the JS team then, thanks :)

**Review by Julien00859 — APPROVED** [review]
Diff LGTM but the commit message is meh :/

The commit tag is `[FIX]`, it is useless to repeat "fix bug" in the title, thanks to the tag we know you are fixing a bug and it would save 8 characters in the title :D

The title states about "searching for a currency", searching where? using `search()` or searching using the list view?

The message is

> Before this commit, searching for a currency in the settings would cause an error.
> This commit fixes this.

Usually we give precise steps explaining how to reproduce the bug, e.g.

> Install account, go to the account settings, click on the "currencies" link. You end up on the list view of currencies. In the control panel, search for anything. JS error: invalid search domain.

If we consider worth it, we continue with a small explanation of the underlying problem, best is if you can locate the root issue. Here the issue mostly is a typo in the commit that introduced the feature. e.g.

> Filter domain in xml views are processed by Javascript which normalize them. The sanitation function explicitly reject non-list domains, including tuples. [...]

With the problem identified comes two possibles solutions: (1) change the filter domain to replace the tuple by a list or (2) challenge the conditional in the `normalizeDomainAST`, determine if tuples should be considered as valid datastructures.

If you go with (1) then the explanation can continue with, e.g.

> [...] The commit 7a4be2f wrongly used a tuple for the domain.

If you go with (2) then the explanation can continue with, e.g.

> [...] Tuples are commonly used in domains python-side and should be accepted by the normalization function JS-side. The normalization function still return the domain as an array.

---

The precise steps to reproduce the problem and the explanation of the underlying source cause are very important. Because you didn't include a unittest, I did fire up a local database and tried to reproduce the problem myself. At first I tried searching for a currency from within the settings and ended up wasting 3m before going to the currency *list view*.

image

Then I scratched my head a couple more minutes to understand why I got that "Invalid search domain" for a tuple and not a list. A quick grep on the error string and a quick breakpoint in that `normalizeDomainAST` function to understand that tuple are `type=10` and list are `type=4`.

Here it was easy to understand what was going on and to review + approve the diff so it is fine :)  Keep in mind that reviewers can be dragged out of nowhere (e.g. here, due of the code-owner file) so when the problem is more subtle, we *really* need those extra bit of info in order to review the changes.

**Comment by rigr-odoo** [comment]
@Julien00859 Thanks for the exhaustive review. I was not expecting so many comments for such a simple PR, but it is indeed better to have a thorough commit with all the info :)

I will update the PR while waiting for JS team's feedback

**Comment by kebeclibre** [comment]
https://github.com/odoo/odoo/pull/97850

**Pull request closed without merging**
